node/pkg/guardiansigner/guardiansigner.go

```diff
@@ -0,0 +1,66 @@
+package guardiansigner
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"strings"
+)
+
+// The types of guardian signers that are supported
+type SignerType int
+
+const (
+	InvalidSignerType SignerType = iota
+	FileSignerType
+)
+
+// GuardianSigner interface
+type GuardianSigner interface {
+	// TODO: document that the signer implementations assume they are recieving a keccack256 hash
+	Sign(hash []byte) (sig []byte, err error)
+	PublicKey() (pubKey ecdsa.PublicKey)
+	Verify(sig []byte, hash []byte) (valid bool, err error)
+}
+
+func NewGuardianSignerFromUri(signerUri string, unsafeDevMode bool) (GuardianSigner, error) {
+
+	// Get the signer type
+	signerType, signerKeyConfig := ParseSignerUri(signerUri)
+
+	switch signerType {
+	case FileSignerType:
+		return NewFileSigner(unsafeDevMode, signerKeyConfig)
+	case InvalidSignerType:
+		return nil, fmt.Errorf("unsupported guardian signer type")
+	default:
+		return nil, fmt.Errorf("unsupported guardian signer type")
+	}
+}
+
+func ParseSignerUri(signerUri string) (signerType SignerType, signerKeyConfig string) {
+	signerUriSplit := strings.Split(signerUri, "://")
+
+	if len(signerUriSplit) < 2 {
+		return InvalidSignerType, ""
+	}
+
+	typeStr := signerUriSplit[0]
+	keyConfig := strings.Join(signerUriSplit[1:], "")
+
+	switch typeStr {
+	case "file":
+		return FileSignerType, keyConfig
+	default:
+		return InvalidSignerType, ""
+	}
+}
+
+// WARNING: DO NOT USE THIS SIGNER OUTSIDE OF TESTS
```

Comment:
Maybe it would be helpful to rename it to something like `GenerateSignerWithPrivatekeyUnsafe` or else move it into a testing library if that's possible.

Reply:
Renamed the function. Will move it to a different file if there are more requests like this.